rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -202,12 +202,24 @@ void Player::play()
       "Invalid delay value: " << play_options_.delay.nanoseconds() << ". Delay is disabled.");
   }
 
+  rcutils_time_point_value_t play_until_time = starting_time_;
+  if (play_options_.playback_duration >= rclcpp::Duration(0, 0)) {
+    play_until_time += play_options_.playback_duration.nanoseconds();
+  } else {
+    play_until_time = -1;
+    RCLCPP_INFO_STREAM(
+      get_logger(),
+      "Invalid playback duration value: " << play_options_.playback_duration.nanoseconds() <<
+        ". Playback duration is disabled.");
+  }
```

Comment:
Nitpick optimization for better readability of the logic;
```suggestion
  rcutils_time_point_value_t play_until_time = -1;
  if (play_options_.playback_duration >= rclcpp::Duration(0, 0)) {
    play_until_time = starting_time_ + play_options_.playback_duration.nanoseconds();
  }
```
I think in this particular  case we agreed and have added notice in header file where `play_options_.playback_duration` defined that any negative value means play until end.  i.e. negative value is a valid case and don't need to print error message in log.